src/d3a/models/state.py

```diff
@@ -43,36 +47,154 @@
 # - If a device has no state, maybe it doesn't need its own appliance class either
 
 
-class PVState:
+class StateInterface(metaclass=ABCMeta):
+    """Interface containing methods that need to be defined by each State class."""
+
+    @abstractmethod
+    def get_state(self) -> Dict:
+        """Return the current state of the device."""
+        return {}
+
+    @abstractmethod
+    def restore_state(self, state_dict: Dict):
+        pass
+
+    @abstractmethod
+    def delete_past_state_values(self, time_slot):
+        pass
+
+
+class ConsumptionState(StateInterface):
```

Comment:
`ConsumptionState` and `ProductionState` both are not implementing `delete_past_state_values`, but they should, right?

Reply:
You're right. Since the implementation of `delete_past_state_values` is different in every device State class, I didn't add it to `Consumption/Production` states. But I could do it since it would make more sense to have it there, rather than on Load/PV states.